eclipse-theia/theia
# add ts-clean script

When doing refactorings, one might end up renaming/deleting TypeScript
source files. But if a build was run previously, the outputted
JavaScript files will still be in the build output folder. Then
extensions that depend on the refactored package will not fail to build
because the old path still resolves to the lingering JavaScript files.

Add `ts-clean` script that deletes all generated files without a
corresponding source file.

Add the missing `compile` script to most packages.

#### How to test

- Build those changes.
- Rename a TypeScript file to something new.
- Run `npx ts-clean dev-packages/* packages/* --dry`.
- The tool should prompt about a missing source file and deleting generated files.

The tool supports a `--watch (-w)` mode, use `--dry` to only get logs without touching the fs.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
